src/convert-base.jl

```diff
@@ -227,7 +227,7 @@ sexp(d::Associative) = sexp(VecSxp,d)
 
 function rcopy{A<:Associative,S<:VectorSxp}(::Type{A}, s::Ptr{S})
     a = A()
-    K,V = eltype(a)
+    K,V = eltype(a).parameters
```

Comment:
I don't think we should be using these internal fields. Better to use:
```
K = keytype(a)
V = valtype(a)
```

Other than that, this looks great, thanks!